h/data_tasks/report/create_from_scratch/08_annotation_counts/01_create_view.sql

```diff
@@ -0,0 +1,55 @@
+-- This is a summary table which re-interprets the annotations data in terms of
+-- weekly counts split by a number of facets:
+--
+--  * Authority
+--  * Group
+--  * User
+--  * Annotation sub-type
+--  * Annotation shared status
+--
+-- This is intended to speed up a variety of summary queries over the table and
+-- make queries about the sub-type easy.
+
+DROP MATERIALIZED VIEW IF EXISTS report.annotation_counts CASCADE;
+
+-- There are various indicators of different sub-types of annotation, only
+-- certain combinations should happen together. This table shows all
+-- combinations and how we will interpret them.
+
+-- This table is in the same order as the CASE statement below.
+
+-- |----------|--------------|-----------|------------|-----------|
+-- | Is Root? | Is Anchored? | Has text? | Sub-Type   | Expected? |
+-- |----------|--------------|-----------|------------|-----------|
+-- | False    | True         | True      | Reply      | No!       |
+-- | False    | True         | False     | Reply      | No!       |
+-- | False    | False        | True      | Reply      | Yes       |
+-- | False    | False        | False     | Reply      | No!       |
+-- | True     | False        | True      | Page-note  | Yes       |
+-- | True     | False        | False     | Page-note  | No!       |
+-- | True     | True         | False     | Highlight  | Yes       |
+-- | True     | True         | True      | Annotation | Yes       |
+-- |----------|--------------|-----------|------------|-----------|
+
+CREATE MATERIALIZED VIEW report.annotation_counts AS (
```

Comment:
This is a nearly exact copy of `annotation_group_type_counts`. I wasn't going to call this `annotation_group_user_public_type_counts`. I could easily be wrong, but I don't know there's much more we can split by that still makes sense before you are back to a plain annotation table. Thus `annotation_counts`.

I'm hoping in future to refactor away some of the other tables which are very similar to this one (or could be derived from it).